### PR TITLE
Drafting xpbd f64 support

### LINKED ISSUE #46: Potential f64 support
Hello!
Thank you for providing a character controller crate!
After seeing this crate for the first time and trying to integrate with a bevy/xpbd setup I had, i came to encounter some compilation issues and found that it seems the crate used `f32` as default from xpbd and rapier.
Hence I was wondering about this issue:

# What

As both Rapier and XPBD provide `f64` precision alternative features(xpbd)/crates([rapier](https://crates.io/crates/rapier2d-f64)). Would it be possible  to adapt this crate to take advantages of those?

# Why

- Higher simulation precision
- Prevent rounding errors due to f32
- ...

### PR DESCRIPTION
**Pull request by vpochapuis**
Hello!

# What

This is a little draft of support for `xpbd` f64 precision.
I would need some help to make sure its coherent as I am not very well versed in this kind of crates.

Resolves #46 

# How

To enable the feature one need to use
> bevy-tnua = {version = "0.15.0", default-features = false,  features = ["f64"]}
> bevy-tnua-xpbd2d = {version = "0.2.0", default-features = false,  features = ["f64"]}

# Known issues

Testing with 

> cargo test --no-default-features --features f64

doesn't work as I can't figure out why the `dev-dependencies` is not taking in account the `default-features = false` and importing `parry` in both `f32` and `f64`.

However the crate works as intended when imported in another project.

# Notes

The `math` module is basically the same as the one in `xpbd`, but i thought that since `xpbd` isn't a dependency when using `rapier`, might not be great to rely on it directly? I am not sure.

### CODE REVIEW

File: physics-integration-layer/src/math.rs
Comment (idanarye) [inline physics-integration-layer/src/math.rs:1]: This is a bit of a nitpick, but why not use the naming convention XPBD uses? (`Float`, `Vector`, etc.)

File: physics-integration-layer/src/math.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    fn adjust_precision(&self) -> Self::Adjusted {
+        #[cfg(feature = "f64")]
+        return *self;
+        #[cfg(not(feature = "f64"))]
+        return self.as_vec2();
+    }
+}
+
+/// Adjust the precision down to `f32` regardless of compilation.
+pub trait AsF32 {
```
Comment (idanarye) [inline physics-integration-layer/src/math.rs:119]: Why not merge this functionality into `AdjustPrecision`? Give it a second method - maybe `bevy_precision` or `lower_precision` - which brings it to the same precision Bevy uses.

File: Cargo.toml [hunk 2]
```diff
@@ -53,8 +53,12 @@ bevy = { version = "^0.13", default-features = false, features = [
...
-bevy_xpbd_3d = { version = "^0.4" }
-bevy-tnua-xpbd3d = { path = "xpbd3d" }
+bevy_xpbd_3d = { version = "^0.4", default-features = false, features = ["3d","debug-plugin", "parallel", "async-collider"] }
+bevy-tnua-xpbd3d = { path = "xpbd3d", default-features = false }
 
 [package.metadata.docs.rs]
 all-features = true
+
+[features]
+default = [  "bevy_xpbd_3d/parry-f32" ]
```
[the remark this answers is not in the document]
Comment (vpochapuis) [reply Cargo.toml]: Thank you very much for the feedback!
I will try to fix those!

[the remark this answers is not in the document]
Comment (vpochapuis) [reply Cargo.toml]: Potentially solved in f6ae729

File: build.rs
[the remark this answers is not in the document]
Comment (vpochapuis) [reply build.rs:1]: Potentially solved in f6ae729

File: physics-integration-layer/src/math.rs
Comment (vpochapuis) [reply physics-integration-layer/src/math.rs:1]: There was no specific reason and I will change it!

File: physics-integration-layer/src/math.rs (hunk 1 above)
Comment (vpochapuis) [reply physics-integration-layer/src/math.rs:119]: I was trying to follow the same way XPBD was using in their math library [here](https://github.com/Jondolf/bevy_xpbd/blob/e9b98d05be9e14086a8b6182d1bb55376cc978e4/src/math/mod.rs#L40), should I still integrate in the same Trait?

File: physics-integration-layer/src/math.rs
Comment (vpochapuis) [reply physics-integration-layer/src/math.rs:1]: Fixed in 50ed22a

File: physics-integration-layer/src/math.rs (hunk 1 above)
Comment (idanarye) [reply physics-integration-layer/src/math.rs:119]: In that case, let's just leave it as that for now.

**Comment by vpochapuis** [comment]
Hello,
For the Ci Docs not passing, it seems `--all-features` doesn't work. As the feature  model is similar as the one used in `bevy-xpbd` the only way to be able to use it would be either using default feature or disabling default and using the f64 one: https://github.com/Jondolf/bevy_xpbd/blob/main/.github/workflows/ci.yml.
I am not sure whether its a good practice or not, I am very eager to have your feedback thank you!

**Comment by idanarye** [comment]
We don't really need the `--all-features` because Tnua does not use features, so it should be fine.

**Pull request merged**